File: Sully.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
int main(){
	char cur[512];
	char file[512];
	char exe[512];
	int i = 5;
	if (strcmp(__FILE__, "Sully.c") != 0)
		i--;
	if (i < 0) return (0);
	sprintf(cur, "Sully_%d.c", i);
	FILE *fd = fopen(cur,"w+");
	char* s = "#include <stdio.h>%1$c#include <stdlib.h>%1$c#include <string.h>%1$cint main(){%1$c%2$cchar cur[512];%1$c%2$cchar file[512];%1$c%2$cchar exe[512];%1$c%2$cint i = %6$d;%1$c%2$cif (strcmp(__FILE__, %3$cSully.c%3$c) != 0)%1$c%2$c%2$ci--;%1$c%2$cif (i < 0) return (0);%1$c%2$csprintf(cur, %3$cSully_%5$s.c%3$c, i);%1$c%2$cFILE *fd = fopen(cur,%3$cw+%3$c);%1$c%2$cchar* s = %3$c%4$s%3$c;%1$c%2$cfprintf(fd,s,10,9,34,s,%3$c%5$s%3$c,i,%3$c%7$s%3$c);%1$c%2$cfclose(fd);%1$c%2$csprintf(file, %3$cclang -Wall -Werror -Wextra -o %7$s%5$s %7$s%3$c, %3$cSully_%3$c, i, cur);%1$c%2$csystem(file);%1$c%2$cif (i == 0) return (0);%1$c%2$csprintf(exe, %3$c./%7$s%5$s%3$c, %3$cSully_%3$c, i);%1$c%2$csystem(exe);%1$c}%1$c";
	fprintf(fd,s,10,9,34,s,"%d",i,"%s");
	fclose(fd);
	sprintf(file, "clang -Wall -Werror -Wextra -o %s%d %s", "Sully_", i, cur);
	system(file);
	if (i == 0) return (0);
	sprintf(exe, "./%s%d", "Sully_", i);
	system(exe);
}

File: PHP/Sully.php
<?php
function foo()
{
	$cur = "NULL";
	$file = "NULL";
	$i = 5;
	if (strcmp(basename(__FILE__), "Sully.php") != 0)
		$i--;
	if ($i < 0) return (0);
	$cur = sprintf("Sully_%d.php", $i);
	$fd = fopen($cur,"w+");
	$s = "<?php%cfunction foo()%c{%c%c%ccur = %cNULL%c;%c%c%cfile = %cNULL%c;%c%c%ci = %d;%c%cif (strcmp(basename(__FILE__), %cSully.php%c) != 0)%c%c%c%ci--;%c%cif (%ci < 0) return (0);%c%c%ccur = sprintf(%cSully_%s.php%c, %ci);%c%c%cfd = fopen(%ccur,%cw+%c);%c%c%cs = %c%s%c;%c%cfprintf(%cfd,%cs,10,10,10,9,36,34,34,10,9,36,34,34,10,9,36,%ci,10,9,34,34,10,9,9,36,10,9,36,10,9,36,34,%c%cd%c,34,36,10,9,36,36,34,34,10,9,36,34,%cs,34,10,9,36,36,36,34,37,34,36,34,37,34,10,9,36,10,9,36,34,%c%cs%c,34,36,10,9,36,10,10,10,10);%c%cfclose(%cfd);%c%c%cfile = sprintf(%cphp %s%c, %ccur);%c%csystem(%cfile);%c}%cfoo();%c?>%c";
	fprintf($fd,$s,10,10,10,9,36,34,34,10,9,36,34,34,10,9,36,$i,10,9,34,34,10,9,9,36,10,9,36,10,9,36,34,"%d",34,36,10,9,36,36,34,34,10,9,36,34,$s,34,10,9,36,36,36,34,37,34,36,34,37,34,10,9,36,10,9,36,34,"%s",34,36,10,9,36,10,10,10,10);
	fclose($fd);
	$file = sprintf("php %s", $cur);
	system($file);
}
foo();
?>
